### ISSUE TITLE
Add in UI support for missing rancher-compose/docker-compose fields

### ISSUE THREAD

**Issue by deniseschannon** [rancher/rancher#6282]
As docker-compose/rancher-compose has added supported fields, we are missing the support within Rancher. 

After Rancher-compose has added support, there are additional areas that need to be updated
1. Adding the field to the container API
2. Adding the field to the exported docker-compose.yml of the stack
3. Adding the field to be read/imported by rancher-compose
   **4. UI (Once Cattle is complete, we can track the UI in a separate issue) - This issue is to track the UI support for the new fields.**

Known Fields that are missing.

| Field | Description |
| --- | --- |
| \- [ ] --add-host=[] | Add a custom host-to-IP mapping (host:ip) |
| \- [ ]   --blkio-weight | Block IO (relative weight), between 10 and 1000 |
| \- [ ]   --blkio-weight-device=[] | Block IO weight (relative device weight) |
| \- [ ]  --cpu-shares | CPU shares (relative weight) |
| \- [ ]  --cgroup-parent | Optional parent cgroup for the container |
| \- [ ]  --cpu-period | Limit CPU CFS (Completely Fair Scheduler) period |
| \- [ ]  --cpu-quota | Limit CPU CFS (Completely Fair Scheduler) quota |
| \- [ ]  --cpuset-cpus | CPUs in which to allow execution (0-3, 0,1) |
| \- [ ]  --cpuset-mems | MEMs in which to allow execution (0-3, 0,1) |
| \- [ ]  --device-read-bps=[] | Limit read rate (bytes per second) from a device |
| \- [ ]  --device-read-iops=[] | Limit read rate (IO per second) from a device |
| \- [ ]  --device-write-bps=[] | Limit write rate (bytes per second) to a device |
| \- [ ]  --device-write-iops=[] | Limit write rate (IO per second) to a device |
| \- [ ]  --dns-opt=[] | Set DNS options |
| \- [ ]  --dns-search=[] | Set custom DNS search domains |
| \- [ ]  --group-add=[] | Add additional groups to join |
| \- [ ]  --ip | Container IPv4 address (e.g. 172.30.100.104) |
| \- [ ]  --ip6 | Container IPv6 address (e.g. 2001:db8::33) |
| \- [ ]  --ipc | IPC namespace to use |
| \- [ ]  --isolation | Container isolation level |
| \- [ ]  --kernel-memory | Kernel memory limit |
| \- [ ]  --memory-reservation | Memory soft limit |
| \- [ ]  --memory-swap | Swap limit equal to memory plus swap: '-1' to enable unlimited swap |
| \- [ ]  --memory-swappiness=-1 | Tune container memory swappiness (0 to 100) |
| \- [ ]  --net-alias=[] | Add network-scoped alias for the container |
| \- [ ]  --oom-kill-disable | Disable OOM Killer |
| \- [ ]  --oom-score-adj | Tune host's OOM preferences (-1000 to 1000) |
| \- [ ]  --read-only | Mount the container's root filesystem as read only |
| \- [ ]  --security-opt=[] | Security Options |
| \- [ ]  --shm-size | Size of /dev/shm, default value is 64MB |
| \- [ ]  --stop-signal=SIGTERM | Signal to stop a container, SIGTERM by default |
| \- [ ]  --sysctl value | Sysctl options (default map[]) |
| \- [ ]  --tmpfs=[] | Mount a tmpfs directory |
| \- [ ]  --ulimit=[] | Ulimit options |
| \- [ ]  --uts | UTS namespace to usea |

**Comment by deniseschannon** [comment]
The original task has been split into 4 issues:

Cattle Changes: #4708
Rancher-compose: #6280
Exporting yml: #6281
UI: #6282

**Comment by ibuildthecloud** [comment]
Adding these field to UI (compose/CLI) should be selective to ones we know work.

**Comment by SamMousa** [comment]
tmpfs works via docker-compose and in the API (the tmpfs system is mounted in the container so I assume cattle also supports it).
Does that mean it can be added to the GUI?

**Comment by vincent99** [comment]
tmpfs and several others of these are in 2.0.  Blanket supporting every field doesn't make sense for the UI, so I'm callign this done and people can make separate arguments for any additional fields for 2.0.

**Comment by davidwindell** [comment]
@vincent99 is shm-size in 2.0?

**Comment by vincent99** [comment]
No, I don't think it's supported in k8s: https://github.com/kubernetes/kubernetes/issues/28272

**Comment by davidwindell** [comment]
Ok thanks, I'll try to use `-v /dev/shm:/dev/shm` to workaround for now.
